src/actions/reports/cost.py

```diff
@@ -614,8 +780,8 @@ def show_charts_and_handle_events(self, spec, fig, axs):
         def on_pick(event):
             ann = anns[id(event.mouseevent.inaxes)]
             series = event.line.get_label()
-            series_data = spec.series_data[series][event.ind[0]]
-            node: PlanNode = series_data.node
+            data_point = spec.series_data[series][event.ind[0]]
+            node: PlanNode = data_point.node
```

Comment:
PlanNode doesn't have `get_search_condition_str` used in 630(796) line. Need to add this method to the PlanNode object.

Reply:
Search conditions are only relevant to the ScanNode. Will add a check for instance type.

Reply:
1. Moved to `ScanNode.__str__` for the use cases that just prints the node info.
2. Created a wrapper that checks the node instance type for the use cases that processes the expression itself, e.g. ChartSpec node filter conditions, etc.